prov/sm2/src/sm2_atomic.c

```diff
@@ -109,6 +109,7 @@ sm2_do_atomic_inject(struct sm2_ep *ep, int64_t peer_gid, uint32_t op,
 		return ret;
 
 	xfer_entry->hdr.proto = sm2_proto_inject;
+	xfer_entry->hdr.proto_flags |= proto_flags;
```

Comment:
the field needs to be reset after the pop - don't OR in, just set

Reply:
We set proto_flags to 0 in generic format, so OR is OK. But it was in the wrong place. Fixed now. https://github.com/ofiwg/libfabric/blob/main/prov/sm2/src/sm2.h#L198

It would've been caught by FT_UNIT_TEST but there are other failures that @a-szegel will work on